library/std/src/panicking.rs

```diff
@@ -644,7 +659,13 @@ fn rust_panic_with_hook(
         if panics > 2 {
             // Don't try to print the message in this case
             // - perhaps that is causing the recursive panics.
-            rtprintpanic!("thread panicked while processing panic. aborting.\n");
+            if must_abort {
```

Comment:
What's the reason for adding this conditional?

Reply:
The only reason is to print out a different message so that the user is aware about the uncertainty. I thought it might be helpful but is not required.

Reply:
Extra message is removed as it is not very helpful and only in very rare cases.

Reply:
From zulip (talking about the case where this branch is hit)

> Thom Chiovoloni: there's a slight race where
>
> - T1 panics, calls increase
> - T2 forks, panics in child before exec
> - T1 (in parent) calls decrease
>
> Thom Chiovoloni: T2 can't look at the local panic count and only global, so it can't if this is different from
> - some thread panics
> - while unwinding, the same thread forks and the child panics before exec
>
> right?

I think this means that we *ideally* would handle this case the `must_abort` case in here the same as below (but without the panic info). In other words: `rtprintpanic!("panicked after panic::always_abort(), aborting.\n");`

Or is there some other case where this matters?

(All that said, I'm kind of fine leaving this with the existing message, since this is *suuuuch* an edge case. That said, giving a slightly better message can't hurt).